processing/src/main/java/org/apache/druid/query/planning/ExecutionVertex.java

```diff
@@ -33,30 +33,61 @@
 import org.apache.druid.query.scan.ScanQuery;
 import org.apache.druid.query.spec.MultipleIntervalSegmentSpec;
 import org.apache.druid.query.spec.QuerySegmentSpec;
+import org.apache.druid.segment.SegmentReference;
+import org.apache.druid.segment.VirtualColumn;
 import org.apache.druid.segment.join.JoinPrefixUtils;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 
 /**
- * Represents the native engine's execution vertex.
+ * Represents the native engine's execution vertex - the execution unit it may execute in one go.
  *
- * Most likely due to evolutional purposes there are some concepts which went
- * beyond their inital design:
+ * This minimal Vertex a single {@link Query} with an input {@link DataSource}.
  *
- * Multiple queries might be executed in one stage: <br/>
- * The GroupBy query could be collapsed at exection time).
+ * However there might be more complicated cases.
  *
- * Dag of datasources: <br/>
- * an execution may process an entire dag of datasource in some cases
- * (joindatasource) ; or collapse some into the execution (filter)
+ * Multiple queries could be executed in one stage: <br/>
+ * the GroupBy query could be collapsed at execution time.
+ *
+ * For example: <br/>
+ * SELECT COUNT(*) FROM (SELECT string_first_added FROM druid.wikipedia_first_last GROUP BY 1)
+ *
+ * Will have 2 groupby queries; but they will be executed together.
```

Comment:
link here to `QueryToolChest#canPerformSubquery`, it would be helpful for readers to see that this is the place that determines what can be executed together.